lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.mel

```diff
@@ -987,27 +1069,35 @@ global proc string addMayaReferenceToUsd(string $ufePath)
         {
             $tmpValue = `optionVar -query usdMayaReferenceVariantSetNameText`;
         }
-        $qVarSetName = quoteString($tmpValue);
+        $varSetName = $tmpValue;
 
         $tmpValue = `optionVar -query usdMayaReferenceVariantNameOption`;
         if ($tmpValue == $createNew)
         {
             $tmpValue = `optionVar -query usdMayaReferenceVariantNameText`;
         }
-        $qVarName = quoteString($tmpValue);
+        $varName = $tmpValue;
     }
 
     int $autoEditAsMayaData = `optionVar -query usdMayaReferenceEditAsMayaDat`;
     
     // No need to actually create the Maya reference node: pulling on the
     // Maya reference prim will create the Maya reference node.
     // Create the Maya reference prim under its parent.
-    string $qUfePath = quoteString($ufePath);
-    string $qPrimName = quoteString($gUsdMayaReferencePrimName);
     string $filePath = fromNativePath($file[0]);
-    string $qFilePath = quoteString($filePath);
-    string $qNs = quoteString(`computeNamespace($filePath)`);
-    python("import mayaUsdAddMayaReference; mayaUsdAddMayaReference.createMayaReferencePrim(" + $qUfePath + ", " + $qFilePath + ", " + $qNs + ", " + $qPrimName + ", " + $qVarSetName + ", " + $qVarName + ", " + $autoEditAsMayaData + ")");
+    string $ns = computeNamespace($filePath);
+    string $pyCmdArgs = `format -s $ufePath -s $filePath -s $ns -s $gUsdMayaReferencePrimName "'^1s', '^2s', '^3s', '^4s'"`;
+    if ($useGroup)
+    {
+        $pyCmdArgs += `format -s $groupName -s $groupType -s $groupKind ", groupPrim=('^1s', '^2s', '^3s')"`;
+    }
+    if (($varSetName != "") && ($varName != ""))
+    {
+        $pyCmdArgs += `format -s $varSetName -s $varName ", variantSet=('^1s', '^2s')"`;
+    }
+    $pyCmdArgs += `format -s $autoEditAsMayaData ", mayaAutoEdit='^1s'"`;
+    string $pyCmd = `format -s $pyCmdArgs "import mayaUsdAddMayaReference; mayaUsdAddMayaReference.createMayaReferencePrim(^1s)"`;
+    python($pyCmd);
```

Comment:
Reworked the calling of createMayaReferencePrim since it now has many optional params.